pkg/controller/servicebinding/secret_test.go

```diff
@@ -37,18 +35,96 @@ func TestSecretNew(t *testing.T) {
 	s := newSecret(
 		f.FakeDynClient(),
 		ns,
-		name,
+		secretName,
+		data,
+		secretOwnerReference,
 	)
 
-	t.Run("createOrUpdate", func(t *testing.T) {
-		u, err := s.createOrUpdate(data, secretOwnerReference)
+	t.Run("unique_secret_name", func(t *testing.T) {
+		hashedName1, err := name(secretName, data)
+		assert.NoError(t, err)
+		assert.Contains(t, hashedName1, secretName)
+
+		hashedName2, err := name(hashedName1, data)
+		assert.NoError(t, err)
+		assert.Equal(t, hashedName2, hashedName1)
+	})
+
+	t.Run("check secret creation", func(t *testing.T) {
+		u, err := s.overwrite()
+		assert.NoError(t, err)
+		assertSecretNamespacedName(t, u, ns, secretName)
+		existingSecret, err := s.get()
 		assert.NoError(t, err)
-		assertSecretNamespacedName(t, u, ns, name)
+		hashedName, err := name(secretName, data)
+		assert.NoError(t, err)
+		assert.Equal(t, hashedName, existingSecret.GetName())
 	})
 
-	t.Run("Get", func(t *testing.T) {
-		u, err := s.get()
+	t.Run("different_secret_payload_are_different_secret", func(t *testing.T) {
+		payload1 := map[string][]byte{"dbName": []byte("postgres")}
+		value1 := newSecret(
+			f.FakeDynClient(),
+			ns,
+			secretName,
+			payload1,
+			secretOwnerReference,
+		)
+		secret1, err := value1.overwrite()
+		assert.NoError(t, err)
```

Comment:
is this needed as the part of the setup? If yes then we should check how many secret really exist in the cluster after applying two times overwrite.